pkg/iac/terraform/state/enumerator/s3.go

```diff
@@ -8,26 +8,26 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/cloudskiff/driftctl/pkg/envproxy"
 	"github.com/pkg/errors"
 
 	"github.com/bmatcuk/doublestar/v4"
 	"github.com/cloudskiff/driftctl/pkg/iac/config"
 )
 
-type S3EnumeratorConfig struct {
-	Bucket *string
-	Prefix *string
-}
-
 type S3Enumerator struct {
 	config config.SupplierConfig
 	client s3iface.S3API
 }
 
 func NewS3Enumerator(config config.SupplierConfig) *S3Enumerator {
+	envProxy := envproxy.NewEnvProxy()
+	envProxy.SetProxy("DCTL_S3_", "AWS_")
+	envProxy.Apply()
 	sess := session.Must(session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	}))
+	envProxy.Restore()
 	return &S3Enumerator{
 		config,
 		s3.New(sess),
```

Comment:
Isn't it possible to add a test case to ensure the session parses values from `DCTL_S3_` env vars ?

Reply:
yes probably, i'll do it later today